Add lua 5.4.6 (#48)
unikraft/dynamic-apps: lang/lua/README.md

> 

Reviewer: Remove empty line

> ## Run lua natively on linux

Reviewer: ```suggestion
## Run Lua Natively on Linux
```

> The binary is located in the /usr/bin directory. Run it with the command:

Reviewer: Add empty line before snippet.

> The binary is located in the /usr/bin directory. Run it with the command:

Reviewer: ```suggestion
The binary is located in the `usr/bin` directory.
Run it with the command:
```

> ```

Reviewer: Add `console` as snippet type.

> $ ./usr/bin/lua

Reviewer: ```suggestion
./usr/bin/lua
```

```

## Run lua with Unikraft

Make sure you have the *run-app-elfloader* repository [run-app-elfloader](https://github.com/unikraft/run-app-elfloader).

> Run the *run.sh* script from the *run-app-elfloader* repository:

Reviewer: ```suggestion
Run the `run.sh` script from the [`run-app-elfloader` repository](https://github.com/unikraft/run-app-elfloader):
```

> ```

Reviewer: Add `console` as snippet type.

Reviewer: Add empty line before snippet.

$ ./run.sh -r ../dynamic-apps/lang/lua/ /usr/bin/lua
```

> Run helloworld.lua:

Reviewer: ```suggestion
Run `helloworld.lua`:

```

> ## Run Lua Natively on Linux

Reviewer: Add the following header / intro

```md
# Lua as Dynamic (PIE) Executable

This directory stores the binary executable (ELF - *Executable and Linking Format*), dynamic library dependencies and support files to run Lua.
The binary is PIE (*Position-Independent Executable*).
All files are stored in their expected location in the Linux filesytem hierarchy.
The aim is to run the executable in any Linux ELF-compatible environment: native Linux, chroot/jail, Docker, Unikraft.

# Lua as Dynamic (PIE) Executable

This directory stores the binary executable (ELF - *Executable and Linking Format*), dynamic library dependencies and support files to run Lua.
The binary is PIE (*Position-Independent Executable*).
All files are stored in their expected location in the Linux filesytem hierarchy.
The aim is to run the executable in any Linux ELF-compatible environment: native Linux, chroot/jail, Docker, Unikraft.

> 

Reviewer: ```suggestion

The binary ELF is `./usr/bin/lua`.

```

## Run Lua Natively on Linux

> The binary is located in the `usr/bin` directory.
> Run it with the command:

Reviewer: ```suggestion
To run the binary ELF natively on Linux, run the command below, while inside this directory (`./lang/lua/`):
```

```console
./usr/bin/lua
```

> 

Reviewer: The above command will use the loader and support libraries on your native system, **not** those in this directory.
Because of that, there may be version incompatibilities that give errors such as those below:

--- (replace with backticks)
./usr/bin/lua: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by ./usr/bin/lua)
./usr/bin/lua: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by ./usr/bin/lua)
---

In that case, run the command below that uses the loader and dynamic libraries from the current directory:

--- (replace with triple backticks console)
./lib64/ld-linux-x86-64.so.2 --library-path ./lib ./usr/bin/lua helloworld.lua
---
